src/main/java/com/flippingutilities/model/HistoryManager.java

```diff
@@ -621,9 +625,227 @@ else if (millisBetweenBuyAndSell < 0)
+	 */
+	private List<PartialOffer> getPartialOffers() {
+		int thisItemsId = compressedOfferEvents.get(0).getItemId();
+		List<PartialOffer> partialOffers = new ArrayList<>();
+		parentCombinationFlips.forEach(cf -> partialOffers.addAll(cf.getChildOffers(thisItemsId)));
+		personalCombinationFlips.forEach(cf -> partialOffers.add(cf.parent));
+		return partialOffers;
+	}
+
+	/**
+	 * Gets all the combination flips reference the offers in idsOfOffersToBeDeleted
+	 */
+	private List<CombinationFlip> getInvalidatedCombinationFlips(Set<String> idsOfOffersToBeDeleted) {
+		List<CombinationFlip> combinationFlips = new ArrayList<>();
+		combinationFlips.addAll(getInvalidatedPersonalCombinationFlips(idsOfOffersToBeDeleted));
+		combinationFlips.addAll(getInvalidatedParentCombinationFlips(idsOfOffersToBeDeleted));
+		return combinationFlips;
+	}
+
+	/**
+	 * Gets the personal combination flips that need to be deleted given they reference
+	 * offers that are about to be deleted.
+	 * @param idsOfOffersToBeDeleted the offers that are going to be deleted
+	 */
+	private List<CombinationFlip> getInvalidatedPersonalCombinationFlips(Set<String> idsOfOffersToBeDeleted) {
+		return personalCombinationFlips.stream().
+				filter(cf -> idsOfOffersToBeDeleted.contains(cf.parent.offer.getUuid())).
+				collect(Collectors.toList());
+
+	}
+
+	/**
+	 * Gets the combination flips that need to be deleted from the parent combination
+	 * flip list given that they reference offers that are about to be deleted.
+	 *
+	 * @param idsOfOffersToBeDeleted ids of offers that are about to be deleted
+	 */
+	private List<CombinationFlip> getInvalidatedParentCombinationFlips(Set<String> idsOfOffersToBeDeleted) {
+		if (compressedOfferEvents.isEmpty()) {
+			return new ArrayList<>();
+		}
+		int thisItemsId = compressedOfferEvents.get(0).getItemId();
+		return parentCombinationFlips.stream().
+				filter(cf -> cf.getChildren().get(thisItemsId).keySet().stream().anyMatch(idsOfOffersToBeDeleted::contains)).
+				collect(Collectors.toList());
+	}
+
+	public void addPersonalCombinationFlip(CombinationFlip combinationFlip) {
+		personalCombinationFlips.add(combinationFlip);
+		offerIdToPartialInParentCombinationFlips = null;
+		offerIdToPartialInPersonalCombinationFlips = null;
+	}
+
+	public void addParentCombinationFlip(CombinationFlip combinationFlip) {
+		parentCombinationFlips.add(combinationFlip);
+		offerIdToPartialInParentCombinationFlips = null;
+	}
+
+	/**
+	 * A combination flip cannot exist in both personal and parent combination flips, but list.remove() won't
+	 * throw an error and its convenient to just attempt to remove from both of them as it absolves
+	 * the caller of having to know where the combination flip is.
+	 */
+	public void deleteCombinationFlipForThisItem(CombinationFlip combinationFlip) {
+		personalCombinationFlips.remove(combinationFlip);
+		parentCombinationFlips.remove(combinationFlip);
```

Comment:
ForThisItem comes into play because unlike `deleteCombinationFlips` this method only deleted combination flips for this item and doesn't delete the referenced combination flip in the other items that also store it.

A combination flip for a Guthan's set exists not only in the Guthan's set history manager, but also in the individual pieces HistoryManager. `deleteCombinationFlips` deletes a combination flip everywhere it is referenced, whereas this one deletes it just in the HistoryManager it is called on.